### PR TITLE
[HPI-557] Handling dead letter queue && add unit test for dead letter queue

### PR DESCRIPTION
**Pull request by WarmVissarutRonaldoDude**
- auto create dead letter queue.
- auto set default redrive policy for using dead letter queue if not set redrive policy yet. (which can manually set redrive policy via queueUrl or setRedrivePolicy method)
- fix filtered queue url with filter out dead letter queue.
- add unit test for handling second level dead letter queue.

### CODE REVIEW

File: lib/sqs.js [hunk 1]
```diff
@@ -114,26 +114,203 @@ export default class Sqs {
...
+     */
+    async setRedrivePolicy(queueUrl, deadQueueUrl, redrivePolicy = {}) {
+        // get queue arn
+        const deadQueueAtt = await this.getQueueAttributesAsync(deadQueueUrl);
+        if (deadQueueAtt &&
+            deadQueueAtt.Attributes &&
+            deadQueueAtt.Attributes.QueueArn) {
+            const policy = Object.assign({
+                deadLetterTargetArn: deadQueueAtt.Attributes.QueueArn,
+                maxReceiveCount: 20,
```
Comment (daimagine) [inline lib/sqs.js]: lets make it configurable dude, thru env variable or the argument

Comment (WarmVissarutRonaldoDude) [reply lib/sqs.js]: Sure!

**Comment by WarmVissarutRonaldoDude** [comment]
@daimagine Now it can configure maxReceiveCount 2 ways
1. process.env.POLICY_MAX_RECEIVE_COUNT
2. override by optional redrivePolicy object (which is override process.env.POLICY_MAX_RECEIVE_COUNT if it set)

**Comment by dio** [comment]
@daimagine let's merge this?

**Comment by daimagine** [comment]
okay pak @dio

**Comment by daimagine** [comment]
Please help to release [v0.1.6](https://github.com/HOOQTV/sqs/releases/tag/v0.1.6) on npm pak @dio 🙏 
thank you

**Pull request merged**